### ISSUE TITLE
/plinth/sys/pagekite/configure error 500

### ISSUE THREAD

**Issue by gjedeer** [freedombox/Plinth#1119]
On a fresh Freedombox system.

```
root@freedombox:/home/gdr# apt-cache policy plinth
plinth:
  Installed: 0.15.3+ds-1
  Candidate: 0.15.3+ds-1
  Version table:
 *** 0.15.3+ds-1 500
        500 http://httpredir.debian.org/debian testing/main amd64 Packages
        100 /var/lib/dpkg/status

```

```
  File "/usr/lib/python3/dist-packages/bootstrapform/templatetags/bootstrap.py", line 83, in render
    return template.render(context)
  File "/usr/lib/python3/dist-packages/django/template/backends/django.py", line 64, in render
    context = make_context(context, request, autoescape=self.backend.engine.autoescape)
  File "/usr/lib/python3/dist-packages/django/template/context.py", line 287, in make_context
    raise TypeError('context must be a dict rather than %s.' % context.__class__.__name__)
TypeError: context must be a dict rather than Context.
[2017-11-06 13:00:50,942] plinth.actions INFO     Executing command - ['/usr/share/plinth/actions/upgrades', 'check-auto']
[2017-11-06 13:00:56,353] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy']
[2017-11-06 13:00:56,445] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy'], , 
[2017-11-06 13:00:58,631] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/firewall', 'get-status']
[2017-11-06 13:01:03,934] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy']
[2017-11-06 13:01:04,025] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy'], , 
[2017-11-06 13:01:05,705] plinth.setup   INFO     Running module setup - monkeysphere
[2017-11-06 13:01:05,705] plinth.setup   INFO     Running install for module - monkeysphere, packages - ['monkeysphere']
[2017-11-06 13:01:05,706] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'update']
[2017-11-06 13:01:06,708] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy']
[2017-11-06 13:01:06,897] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy'], , 
[2017-11-06 13:01:07,153] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'install', 'monkeysphere', 'monkeysphere']
[2017-11-06 13:01:10,077] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy']
[2017-11-06 13:01:13,506] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/packages', 'is-package-manager-busy']
[2017-11-06 13:01:16,744] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/monkeysphere', 'host-show-keys']
[2017-11-06 13:01:22,691] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/monkeysphere', 'host-import-key', 'SHA256:2IHHFcPkD+HXNMnVnigU1By1Zd/67CAPyHzQxFTSk/A']
[2017-11-06 13:01:22,836] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/monkeysphere', 'host-import-key', 'SHA256:2IHHFcPkD+HXNMnVnigU1By1Zd/67CAPyHzQxFTSk/A'], , Traceback (most recent call last):
  File "/usr/share/plinth/actions/monkeysphere", line 277, in <module>
    main()
  File "/usr/share/plinth/actions/monkeysphere", line 273, in main
    subcommand_method(arguments)
  File "/usr/share/plinth/actions/monkeysphere", line 215, in subcommand_host_import_key
    key['available_domains'][0]], env=env)
IndexError: list index out of range

[2017-11-06 13:01:22,848] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/monkeysphere', 'host-show-keys']
[2017-11-06 13:09:40,578] axes.watch_login INFO     AXES: Calling decorated function: SSOLoginView
[2017-11-06 13:09:40,582] axes.watch_login INFO     AXES: Calling decorated function: bound_func
[2017-11-06 13:09:43,764] axes.watch_login INFO     AXES: Calling decorated function: SSOLoginView
[2017-11-06 13:09:43,768] axes.watch_login INFO     AXES: Calling decorated function: bound_func
[2017-11-06 13:09:43,869] axes.watch_login INFO     AXES: Successful login by {user: 'gdr', ip: '192.168.2.3', user-agent: 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/62.0.3202.75 Chrome/62.0.3202.75 Safari/537.36', path: '/accounts/login/'}. Creating access record.
[2017-11-06 13:09:43,887] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/auth-pubtkt', 'generate-ticket', '--uid', 'gdr', '--private-key-file', '/etc/apache2/auth-pubtkt-keys/privkey.pem', '--tokens', 'admin']
[2017-11-06 13:09:44,040] axes.watch_login INFO     AXES: Successful login by {user: 'gdr', ip: '192.168.2.3', user-agent: 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/62.0.3202.75 Chrome/62.0.3202.75 Safari/537.36', path: '/accounts/login/'}. Creating access record.
[2017-11-06 13:09:56,931] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/pagekite', 'is-disabled']
[2017-11-06 13:09:57,266] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/pagekite', 'get-kite']
[2017-11-06 13:09:57,593] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/pagekite', 'get-frontend']
[2017-11-06 13:09:57,932] django.request ERROR    Internal Server Error: /plinth/sys/pagekite/configure/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 217, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 215, in _get_response
    response = response.render()
  File "/usr/lib/python3/dist-packages/django/template/response.py", line 107, in render
    self.content = self.rendered_content
  File "/usr/lib/python3/dist-packages/django/template/response.py", line 84, in rendered_content
    content = template.render(context, self._request)
  File "/usr/lib/python3/dist-packages/django/template/backends/django.py", line 66, in render
    return self.template.render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 207, in render
    return self._render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 199, in _render
    return self.nodelist.render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/lib/python3/dist-packages/django/template/loader_tags.py", line 177, in render
    return compiled_parent._render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 199, in _render
    return self.nodelist.render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/lib/python3/dist-packages/django/template/loader_tags.py", line 72, in render
    result = block.nodelist.render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/lib/python3/dist-packages/django/template/loader_tags.py", line 72, in render
    result = block.nodelist.render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 1040, in render
    output = self.filter_expression.resolve(context)
  File "/usr/lib/python3/dist-packages/django/template/base.py", line 736, in resolve
    new_obj = func(obj, *arg_vals)
  File "/usr/lib/python3/dist-packages/bootstrapform/templatetags/bootstrap.py", line 46, in bootstrap_horizontal
    return render(element, markup_classes)
  File "/usr/lib/python3/dist-packages/bootstrapform/templatetags/bootstrap.py", line 83, in render
    return template.render(context)
  File "/usr/lib/python3/dist-packages/django/template/backends/django.py", line 64, in render
    context = make_context(context, request, autoescape=self.backend.engine.autoescape)
  File "/usr/lib/python3/dist-packages/django/template/context.py", line 287, in make_context
    raise TypeError('context must be a dict rather than %s.' % context.__class__.__name__)
TypeError: context must be a dict rather than Context.
```

**Comment by JoKeyser** [comment]
Thank you for reporting! Looks like a duplicate of #1036 ...?

**Comment by JosephKiranBabu** [comment]
This part seems new though.

```
[2017-11-06 13:01:22,836] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/monkeysphere', 'host-import-key', 'SHA256:2IHHFcPkD+HXNMnVnigU1By1Zd/67CAPyHzQxFTSk/A'], , Traceback (most recent call last):
  File "/usr/share/plinth/actions/monkeysphere", line 277, in <module>
    main()
  File "/usr/share/plinth/actions/monkeysphere", line 273, in main
    subcommand_method(arguments)
  File "/usr/share/plinth/actions/monkeysphere", line 215, in subcommand_host_import_key
    key['available_domains'][0]], env=env)
IndexError: list index out of range
```

**Comment by JosephKiranBabu** [comment]
`TypeError: context must be a dict rather than Context.`
This error is fixed in python-django-bootstrap-form 3.3-1
https://tracker.debian.org/news/894656


`IndexError: list index out of range` 
This error has a duplicate. #429 
Hence closing this issue.
